cql3/user_types.cc

```diff
@@ -229,6 +230,17 @@ cql3::raw_value_view user_types::delayed_value::bind_and_get(const query_options
     return options.make_temporary(cql3::raw_value::make_value(user_type_impl::build_value(bind_internal(options))));
 }
 
+shared_ptr<terminal> user_types::marker::bind(const query_options& options) {
+    auto value = options.get_value_at(_bind_index);
+    if (value.is_null()) {
+        return nullptr;
+    }
+    if (value.is_unset_value()) {
+        return constants::UNSET_VALUE;
+    }
```

Comment:
Ah, I see an answer to an earlier question, so I guess github, in its wisdom, decided to reorder the patches so it is impossible for me to understand anything.

Reply:
So again, can we use bind_and_get() (in the call site) instead of bind() to avoid the extra allocation?

If the other setters suffer from the same problem then we can fix everything later.

Reply:
`user_types::marker::bind_and_get` will call `bind`, and then `get` as it currently is implemented (inherited from `non_terminal`). See also my monologue about bugs below your other bind_and_get comment.

We could try to implement an optimized version of `bind_and_get`, but using it in `setter` would only make the `setter` more complicated: `bind_and_get` returns a `bytes` blob, while I want to have an easy access to the separate fields of my UDT (which `multi_item_terminal::get_elements` nicely provides).